commons-math/src/main/java/org/apache/commons/math4/analysis/interpolation/FieldHermiteInterpolator.java

```diff
@@ -0,0 +1,210 @@
+import org.apache.commons.math4.util.MathUtils;
+
+/** Polynomial interpolator using both sample values and sample derivatives.
+ * <p>
+ * The interpolation polynomials match all sample points, including both values
+ * and provided derivatives. There is one polynomial for each component of
+ * the values vector. All polynomials have the same degree. The degree of the
+ * polynomials depends on the number of points and number of derivatives at each
+ * point. For example the interpolation polynomials for n sample points without
+ * any derivatives all have degree n-1. The interpolation polynomials for n
+ * sample points with the two extreme points having value and first derivative
+ * and the remaining points having value only all have degree n+1. The
+ * interpolation polynomial for n sample points with value, first and second
+ * derivative for all points all have degree 3n-1.
+ * </p>
+ *
+ * @param <T> Type of the field elements.
+ *
+ * @since 3.2
+ */
+public class FieldHermiteInterpolator<T extends FieldElement<T>> {
+
+    /** Sample abscissae. */
+    private final List<T> abscissae;
+
+    /** Top diagonal of the divided differences array. */
+    private final List<T[]> topDiagonal;
+
+    /** Bottom diagonal of the divided differences array. */
+    private final List<T[]> bottomDiagonal;
+
+    /** Create an empty interpolator.
+     */
+    public FieldHermiteInterpolator() {
+        this.abscissae      = new ArrayList<>();
+        this.topDiagonal    = new ArrayList<>();
+        this.bottomDiagonal = new ArrayList<>();
+    }
+
+    /** Add a sample point.
+     * <p>
+     * This method must be called once for each sample point. It is allowed to
+     * mix some calls with values only with calls with values and first
+     * derivatives.
+     * </p>
+     * <p>
+     * The point abscissae for all calls <em>must</em> be different.
+     * </p>
+     * @param x abscissa of the sample point
+     * @param value value and derivatives of the sample point
+     * (if only one row is passed, it is the value, if two rows are
+     * passed the first one is the value and the second the derivative
+     * and so on)
+     * @exception ZeroException if the abscissa difference between added point
+     * and a previous point is zero (i.e. the two points are at same abscissa)
+     * @exception MathArithmeticException if the number of derivatives is larger
+     * than 20, which prevents computation of a factorial
+     * @throws DimensionMismatchException if derivative structures are inconsistent
+     * @throws NullArgumentException if x is null
+     */
+    @SafeVarargs
+    public final void addSamplePoint(final T x, final T[] ... value)
+        throws ZeroException, MathArithmeticException,
+               DimensionMismatchException, NullArgumentException {
+
+        MathUtils.checkNotNull(x);
+        T factorial = x.getField().getOne();
+        for (int i = 0; i < value.length; ++i) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;value&#39; (index: 1 | type: T[][). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;x&#39;. The same parameter type is validated here for example: commons-math/src/main/java/org/apache/commons/math4/linear/BlockFieldMatrix.java:152. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).